### DISCUSSION TITLE
Baselining a Supabase Postgres DB with Prisma

### DISCUSSION THREAD

**Discussion by christing12** [prisma/prisma#17988] [Q&A]
Hello! I'm pretty beginner with Postgres /  Supabase / Prisma, and while trying to set up database migrations in prisma I'm running into some issues from a default local supabase database so I'm trying to see if anyone has run into this + has a solution:

After installing all the necessary packages with `npm`  and setting up the `DATABASE_URL` I ran `npx prisma db pull` the following `prisma.schema`: 
```
// prisma.schema
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["multiSchema"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  schemas  = ["auth", "public"]
}
```
after that I followed the prisma guide about baselining the database, so I ran `npx prisma migrate diff --from-empty --to-schema-datamodel prisma/schema.prisma --script > prisma/migrations/0_init/migration.sql` to get the `.sql` file and marked it as applied using `npx prisma migrate resolve --applied 0_init` . That all works fine and it marks the migration as applied. Next I added a basic model to the schema:
```
model authors {
  author_id String @id

  @@schema("public")
}
```
When I try to run `npx prisma migrate dev` after adding this to the schema,  it gets this issue: `Migration 0_init failed to apply cleanly to the shadow database. 
Error:
db error: ERROR: cannot use column reference in DEFAULT expression` . After searching a bit I believe its related to `confirmed_at` and `email` columns  in the followin auto-generated models from supabase: 

```
model identities {
...somefields
  identity_data   Json
  email           String?   @default(dbgenerated("lower((identity_data ->> 'email'::text))"))

  @@id([provider, id])
  @@index([email])
  @@index([user_id])
  @@schema("auth")
}

model users {
 ...somefields
  email_confirmed_at          DateTime?     @db.Timestamptz(6)
  phone_confirmed_at          DateTime?     @db.Timestamptz(6)
  confirmed_at                DateTime?     @default(dbgenerated("LEAST(email_confirmed_at, phone_confirmed_at)")) @db.Timestamptz(6)

  @@index([instance_id])
  @@schema("auth")
}
```

I've tried adding single quotes and double quotes around the relevant fields in `dbgenerated` (ie. `email_confirmed_at` and `identity_data`) but it doesn't fix things. Specifically, when changing it to single quotes `identity_data ->> 'email'::text` generates this error: `db error: ERROR: operator is not unique: unknown ->> text` . Any ideas on how to fix this? I do need the user model because I reference the userID inside of it in another model, so I'm just wondering what my options may be in this case! Here is what running pg_dump on the supabase db returns for those two columns:
```
    email text GENERATED ALWAYS AS (lower((identity_data ->> 'email'::text))) STORED

    confirmed_at timestamp with time zone GENERATED ALWAYS AS (LEAST(email_confirmed_at, phone_confirmed_at)) STORED,
```

**Comment by nurul3101** [comment]
Hello @christing12 👋 

I believe it's a known issue and previously reported here:
- https://github.com/prisma/prisma/issues/17313

This is reported by one of our engineering team members, so I can definitely say that this is on our radar and we do plan to have a solution for this.

**Comment by Devr-pro** [comment]
Having the same issue , it partially blocks the usage & features of prisma , hope it fixes soon

**Reply by nurul3101** [reply]
Our engineering team is aware of this and will try their best to get the issue resolved.

**Reply by Hazemkak** [reply]
@nurul3101 any update on that issue please ?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
